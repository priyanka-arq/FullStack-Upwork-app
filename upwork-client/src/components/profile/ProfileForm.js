import TagsInput from "react-tagsinput";
import AuthService from "../../services/auth.service";
import React, { Component } from "react";

class ProfileForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageUrl: this.props.profile ? this.props.profile.imageUrl : "",
      name: this.props.profile ? this.props.profile.name : "",
      skills: this.props.profile ? this.props.profile.skills : [],
      title: this.props.profile ? this.props.profile.title : "",
      phone: this.props.profile ? this.props.profile.phone : "",
      email: this.props.profile ? this.props.profile.email : "",
      aboutme: this.props.profile ? this.props.profile.aboutme : "",
      suburb: this.props.profile ? this.props.profile.suburb : "",
      postcode: this.props.profile ? this.props.profile.postcode : "",
      resume: this.props.profile ? this.props.profile.resume : "",
      portfolio: this.props.profile ? this.props.profile.portfolio : "",
      linkedIn: this.props.profile ? this.props.profile.linkedIn : "",
      github: this.props.profile ? this.props.profile.github : "",
      redirect: this.props.profile ? this.props.profile.redirect : null,
      previewSource: "",
      currentUser: {},
      userReady: false,
    };
  }

  componentDidMount() {
    const currentUser = AuthService.getCurrentUser();
    this.setState({ currentUser: currentUser, userReady: true });
  }

  handleImageUrl = (event) => {
    this.state.imageUrl = event.target.value;
    this.setState({ imageUrl: event.target.files[0] });
    const file = event.target.files[0];
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onloadend = () => {
      this.setState({ previewSource: reader.result });
    };
    this.setState({ imageUrl: event.target.value });
    //previewFile(this.state.imageUrl);
  };

  onSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit({
      imageUrl: this.state.imageUrl,
      name: this.state.name,
      skills: this.state.skills,
      title: this.state.title,
      phone: this.state.phone,
      email: this.state.email,
      aboutme: this.state.aboutme,
      postcode: this.state.postcode,
      resume: this.state.resume,
      portfolio: this.state.portfolio,
      github: this.state.github,
      redirect: this.state.redirect,
      previewSource: this.state.previewSource,
      user: this.state.currentUser ? this.state.currentUser.id : undefined,
    });
  };

  render() {
    //const { currentUser } = this.state;
    return (
      <div className="form-group">
        <form onSubmit={this.onSubmit}>
          <h3>Name</h3>
          <input
            id="name-input"
            className="form-control"
            onChange={(e) => this.setState({ name: e.target.value })}
            value={this.state.name}
            placeholder="full name"
            required
          />

          <h3>Profile Picture</h3>
          <input
            className="form-control"
            onChange={this.handleImageUrl}
            type="file"
            name="image"
            alt="imageUrl"
          />

          <h3>Title</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ title: e.target.value })}
            value={this.state.title}
            type="text"
            placeholder="title"
            required
          />

          <h3>About me</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ aboutme: e.target.value })}
            value={this.state.aboutme}
            type="text"
            placeholder="about me"
            required
          />

          <h3>Skills</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ skills: e.target.value })}
            value={this.state.skills}
            type="text"
            placeholder="Skills "
            required
          />
          <h3>Phone number:</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ phone: e.target.value })}
            value={this.state.phone}
            type="number"
            name="phone"
            pattern="/(7|8|9)\d{9}/"
            placeholder="Phone Number "
            required
          />
          <h3>Email</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ email: e.target.value })}
            value={this.state.email}
            type="text"
            placeholder="email "
            required
          />

          <h3>Suburb</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ suburb: e.target.value })}
            value={this.state.suburb}
            type="text"
            placeholder="Suburb"
          />

          <h3>Postcode</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ postcode: e.target.value })}
            value={this.state.postcode}
            type="text"
            placeholder="Postcode"
          />

          <h3>Upload Your Resume</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ resume: e.target.value })}
            value={this.state.resume}
            type="file"
            name="resume"
            alt="resume"
            placeholder="Resume"
          />

          <h3>Connect with LinkedIn</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ linkedIn: e.target.value })}
            value={this.state.linkedIn}
            type="text"
            placeholder="LinkedIn"
          />

          <h3>Link to Portfolio</h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ portfolio: e.target.value })}
            value={this.state.portfolio}
            type="text"
            placeholder="Portfolio"
          />

          <h3>Github </h3>
          <input
            className="form-control"
            onChange={(e) => this.setState({ imageUrl: e.target.value })}
            value={this.state.github}
            type="text"
            placeholder="Portfolio"
          />
          <button className="button">Submit</button>
        </form>
        {this.state.previewSource && (
          <img
            src={this.state.previewSource}
            alt="chosen"
            style={{ height: "300px" }}
          />
        )}
      </div>
    );
  }
}

export default ProfileForm;
