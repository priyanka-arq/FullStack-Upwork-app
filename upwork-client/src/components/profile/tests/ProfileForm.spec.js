import React from "react";
import { shallow } from "enzyme";
import ProfileForm from "../ProfileForm";
import profiles from "../../../fixtures/profiles";
import toJSON from "enzyme-to-json";

let wrapped, onSubmitSpy, currentUser, props;

beforeEach(() => {
  // currentUser = { id: 1 };
  onSubmitSpy = jest.fn();

  wrapped = shallow(
    <ProfileForm
      // currentUser={currentUser}
      profile={profiles[0]}
      onSubmit={onSubmitSpy}
    />
  );
});

afterEach(() => {
  wrapped.unmount();
});

test("should render Profile form correctly", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});

describe("the name input ", () => {
  it("should set name on input change", () => {
    wrapped.find("#name-input").simulate("change", {
      target: { value: "Priyanka" },
    });
    expect(wrapped.find("#name-input").prop("value")).toEqual("Priyanka");
    wrapped.update();
  });

  it("should set title on input change", () => {
    const value = "Junior Developer";
    wrapped.find("input").at(2).simulate("change", {
      target: { value },
    });
    expect(wrapped.state("title")).toEqual(value);
  });

  test("should call onSubmit prop for valid form submission", () => {
    wrapped.find("form").simulate("submit", {
      preventDefault: () => {},
    });
    expect(onSubmitSpy).toHaveBeenCalledTimes(1);
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
      imageUrl: profiles[0].imageUrl,
      name: profiles[0].name,
      skills: profiles[0].skills,
      title: profiles[0].title,
      phone: profiles[0].phone,
      email: profiles[0].email,
      aboutme: profiles[0].aboutme,
      postcode: profiles[0].postcode,
      resume: profiles[0].resume,
      portfolio: profiles[0].portfolio,
      github: profiles[0].github,
      previewSource: profiles[0].previewSource,
      // user: currentUser.id,
    });
  });
});
