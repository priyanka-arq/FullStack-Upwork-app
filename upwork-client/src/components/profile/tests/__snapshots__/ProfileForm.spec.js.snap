// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Profile form correctly 1`] = `
<div
  className="form-group"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Name
    </h3>
    <input
      className="form-control"
      id="name-input"
      onChange={[Function]}
      placeholder="full name"
      required={true}
      value="Test"
    />
    <h3>
      Profile Picture
    </h3>
    <input
      alt="imageUrl"
      className="form-control"
      name="image"
      onChange={[Function]}
      type="file"
    />
    <h3>
      Title
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="title"
      required={true}
      type="text"
      value="Junior Developer"
    />
    <h3>
      About me
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="about me"
      required={true}
      type="text"
      value="I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives."
    />
    <h3>
      Skills
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Skills "
      required={true}
      type="text"
      value={
        Array [
          "React",
          "Node",
          "Java",
          "CSS",
        ]
      }
    />
    <h3>
      Phone number:
    </h3>
    <input
      className="form-control"
      name="phone"
      onChange={[Function]}
      pattern="/(7|8|9)\\\\d{9}/"
      placeholder="Phone Number "
      required={true}
      type="number"
      value="04444444444"
    />
    <h3>
      Email
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="email "
      required={true}
      type="text"
      value="test@ga.com"
    />
    <h3>
      Suburb
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Suburb"
      type="text"
      value="Sydney"
    />
    <h3>
      Postcode
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Postcode"
      type="text"
      value={2000}
    />
    <h3>
      Upload Your Resume
    </h3>
    <input
      alt="resume"
      className="form-control"
      name="resume"
      onChange={[Function]}
      placeholder="Resume"
      type="file"
      value=""
    />
    <h3>
      Connect with LinkedIn
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="LinkedIn"
      type="text"
      value="https://linkedin.com/in/chi-mai-nguyenn"
    />
    <h3>
      Link to Portfolio
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Portfolio"
      type="text"
      value="https://little-chi-mai.github.io/portfolio"
    />
    <h3>
      Github 
    </h3>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Portfolio"
      type="text"
      value="https://github.com/little-chi-mai"
    />
    <button
      className="button"
    >
      Submit
    </button>
  </form>
</div>
`;
